aries_vcx/src/handlers/proof_presentation/prover.rs

```diff
@@ -86,30 +92,30 @@ impl Prover {
         Ok(self.prover_sm.get_presentation_msg()?.to_owned())
     }
 
-    pub fn set_presentation(&mut self, presentation: Presentation) -> VcxResult<()> {
-        trace!("Prover::set_presentation >>>");
-        self.prover_sm = self.prover_sm.clone().set_presentation(presentation)?;
-        Ok(())
-    }
-
-    pub async fn send_proposal(
+    pub async fn build_presentation_proposal(
         &mut self,
         proposal_data: PresentationProposalData,
-        send_message: SendClosure,
-    ) -> VcxResult<()> {
-        trace!("Prover::send_proposal >>>");
+    ) -> VcxResult<ProposePresentation> {
+        trace!("Prover::build_presentation_proposal >>>");
         self.prover_sm = self
             .prover_sm
             .clone()
-            .send_presentation_proposal(proposal_data, send_message)
+            .build_presentation_proposal(proposal_data)
             .await?;
-        Ok(())
+        self.prover_sm.get_presentation_proposal()
     }
 
-    pub async fn send_presentation(&mut self, send_message: SendClosure) -> VcxResult<()> {
-        trace!("Prover::send_presentation >>>");
-        self.prover_sm = self.prover_sm.clone().send_presentation(send_message).await?;
-        Ok(())
+    pub async fn mark_presentation_sent(&mut self) -> VcxResult<AriesMessage> {
```

Comment:
need not to be `async` anymore

Reply:
same for underlying `mark_presentation_sent`

Reply:
Removed